app/app.js

```diff
@@ -456,8 +456,12 @@ YUI.add('juju-gui', function(Y) {
             credentials['user-' + envOptions.user] = envOptions.password;
             state.set('authorizedUsers', credentials);
           }
-          envOptions.conn = new sandboxModule.ClientConnection(
-              {juju: new sandboxModule.GoJujuAPI({state: state})});
+          envOptions.conn = new sandboxModule.ClientConnection({
+            juju: new sandboxModule.GoJujuAPI({
+              state: state,
+              socket_url: envOptions.socket_url
```

Comment:
It has an invalid url and any request tries to complete 3 times then fails and throws a notification to the user.